libsolidity/inlineasm/AsmAnalysis.cpp

```diff
@@ -406,7 +406,6 @@ bool AsmAnalyzer::expectDeposit(int const _deposit, int const _oldHeight, Source
 
 Scope& AsmAnalyzer::scope(Block const* _block)
 {
-	auto scopePtr = m_info.scopes.at(_block);
-	solAssert(scopePtr, "Scope requested but not present.");
```

Comment:
This assert checks that `*scopePtr` is a valid operation, this is still a different check than below.